src/Language/Haskell/To/Elm.hs

```diff
@@ -711,7 +719,25 @@ instance (HasElmType a, KnownNat numParams, SOP.HasDatatypeInfo a, SOP.All2 (Has
                     (Expression.App "Json.Encode.string" $ Expression.String $ constructorJSONName constr)
               ]
 
-          _ -> error "Only the DataAeson.TaggedObject sumEncoding is currently supported"
+          Aeson.ObjectWithSingleField ->
+            Expression.Case expr
+              [ case constrShape of
+                  ConstructorShape constrFields ->
+                    ( Pattern.Con (elmConstr constr) (Pattern.Var . fst <$> zip [0..] constrFields)
+                    , Bound.toScope $
+                      Expression.App "Json.Encode.object" $
+                      Expression.List
+                        [ Expression.tuple
+                          (Expression.String $ constructorJSONName constr)
+                          (encodeConstructorFields constrFields)
+                        | not $ null constrFields
+                        ]
+                    )
+                  RecordConstructorShape _recordFields -> error "TODO"
```

Comment:
I'm not quite sure what the `ConstructorShape` type represents, and thus, how exactly this code path is triggered.

Reply:
It represents whether the constructor is a record constructor (`C { a :: A, b :: B }`) or ordinary constructor (`C A B`). I'm not sure if `ObjectWithSingleField` applies in that situation or not so we'd have to check what Aeson does. But I'm guessing the case should be covered.